### PR TITLE
Jetpack Connect: change position of “start with free” button

### LINKED ISSUE #28307: JPC: move “Start with free” button above the fold
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/jetpack/connect/store
2. Try to click “Start with free”.
3. Button is hidden below the fold.

See p6TEKc-2o1-p2

screen-shot-2018-11-01-at-12-50-55-pm

#### What I expected

The free option to be easily visible no matter what screen dimension I'm using. Let's move the button to the top of the page and rename it to “Continue with Free”.

### PR DESCRIPTION
**Pull request by keoshi**
#### Changes proposed in this Pull Request

* Bring the “Start with free” button above the fold by moving it to the top of the page.
* Adds AB test to validate performance depending on button position.
* Relevant e2e tests: https://github.com/Automattic/wp-e2e-tests/pull/1711

#### Testing instructions

1. Starting at URL: `jetpack/connect/store`
2. Check position of “Start with free” button.

#### Before

image

#### After

image

Fixes #28307

### CODE REVIEW

**Comment by keoshi** [comment]
Tests are failing and need updating. Help, Luna!

**Comment by sirreal** [comment]
@keoshi With failing snapshot tests, you've either introduced a regression (doesn't sound like the case) or you'll need to update the snapshots.

We have a guide on that, here's the relevant section: https://github.com/Automattic/wp-calypso/blob/master/docs/testing/snapshot-testing.md#working-with-snapshots

TL;DR: I think this is the shorted command, but check the docs if it doesn't work `npm run test-client -- -u  jetpack-connect`

Run the command to update snapshots then commit and push those changes 👍

**Comment by rickybanister** [comment]
I think this can be a good first test, but I'm curious if it's possible to update the copy above the button to talk about what free offers and how it's sort of 'included'. If I understood correctly that was @joanrho's suggestion. A little more context will help explain that button I think.

**Comment by keoshi** [comment]
@rickybanister Yeah, we can do that here: https://github.com/Automattic/wp-calypso/blob/de20d2755cb3df6b0bb83f610cf762e4970fa206/client/jetpack-connect/plans-grid.jsx#L35-L45

I just wanted to get something out before my day ended.

**Comment by keoshi** [comment]
Thanks @tyxla — I'll fix that last one today.

> Do we want to introduce this change for .com plans as well?

I left .com out of this on purpose so we can move faster.

> How are we planning to measure the results of this? Should we abtest it?

We're planning on having a baseline before the change, and look at the same criteria post-launch. AB testing this would probably be a sensible thing to do. cc @rickybanister @joanrho for thoughts.

**Comment by sirreal** [comment]
For e2e tests to work this needs a rebase. I believe it's branched from a commit that used a bad (now unpublished) package version.

**Comment by keoshi** [comment]
Thanks @sirreal, will take care of that now.

**Comment by keoshi** [comment]
Sorry @sirreal, can you check if I did that correctly? Not sure if I know how to do a rebase, and may have done 💩

**Comment by simison** [comment]
This should do:

```
git rebase origin/master
git push origin update/jetpack-connect-start-free-button-position --force
```

**Comment by keoshi** [comment]
I'm trying to figure this one out with Mikael on Slack.

Leaving his tips here as a note to self:

- `git rebase --abort`
- `git rebase origin/master`
- Get a conflict in `plans-grid.jsx`, solve that conflict and save file.*
- `git add client/jetpack-connect/plans-grid.jsx`
- `git rebase --continue`
- `git push origin update/jetpack-connect-start-free-button-position --force`

* The problem seems to be that there aren't any changes to report on this change.

**Comment by rickybanister** [comment]
We don't want to do this same change for .com, their plans grid is much different now with the launch of the blogger and ecommerce plans. They continue to show the free plan in the grid vs a button. 

If someone can help I'd be happy to run this as an AB test. That would make it easy to track the difference I'd imagine. We want to monitor more than just revenue though, as the old experience results in confusion and cart abandonment.

**Comment by keoshi** [comment]
Thanks @tyxla — agree with you on both points. And not only is it too obvious and makes it very easy to dismiss the plans, the resulting layout is very poor as well.

Will handle the AB portion later this week (I've done it before on https://github.com/Automattic/wp-calypso/pull/23454), but I might ping you anyway! :)

**Comment by keoshi** [comment]
@tyxla No rush, but AB test implementation should be good for another review when you can.

**Comment by keoshi** [comment]
That's intentional, @tyxla — we want to add more spacing below in the top version to set a better rhythm that avoids clashing with the segment buttons.

**Comment by tyxla** [comment]
Seems like we introduced a regression here, making the free plan not work, because the we forgot to move its click handler to the plans grid. Fix suggested in https://github.com/Automattic/wp-calypso/pull/29436.
